book/special.qmd

```diff
@@ -813,15 +813,17 @@ By plotting the result of PCA, we see that our model has separated observations
 
 ```{r special-039, message=FALSE, warning=FALSE}
 task = mlr_tasks$get("usarrests")
+learner = lrn("clust.kmeans")
 prediction = learner$train(task)$predict(task)
 autoplot(prediction, task, type = "pca")
 ```
 
 Silhouette plots visually assess the quality of the estimated clusters by visualising if observations in a cluster are well-placed both individually and as a group.
-The average silhouette index is 0.57, which means observations that are scored below this line are considered to be poor estimations whereas observations above are well clustered.
-To visualise this, silhouette plots include a dotted line at 0.57 and a bar for the score of each data point coloured by cluster.
-
-Continuing with our new example, we find that a lot of observations are actually below the average line, and therefore the quality of our cluster assignments is not very good, meaning that many observations are likely assigned to the wrong cluster.
+To visualise this, silhouette plots include a dotted line at average silhouette coefficient value and a bar for the score of each data point coloured by cluster.
+In our particular case, the average silhouette index is `r round(msr("clust.silhouette")$score(prediction, task), 2)`.
+If many or all data points in a cluster are scored below the average silhouette coefficient line or are near zero then the these observations are considered to be poorly clustered whereas observations with higher silhouette values are well clustered.
```

Comment:
I think intuitively this is still complicated. You are saying a cluster is 'poor' based on the average which itself will depend on those 'poor' clusters! I wonder if something like below taken from [this link](https://www.scikit-yb.org/en/latest/api/cluster/silhouette.html) could be useful?

> In SilhouetteVisualizer plots, clusters with higher scores have wider silhouettes, but clusters that are less cohesive will fall short of the average score across all clusters, which is plotted as a vertical dotted red line.

Reply:
Addressed in the last commit. Thank you!